tests/test_core/test_mypy.py

```diff
@@ -0,0 +1,33 @@
+import boto3
+
+from moto import mock_s3
+from moto.core.models import BaseMockAWS
+
+
+@mock_s3
+def test_without_parentheses() -> int:
```

Comment:
@bblommers The name of this test is from your original draft MR, but I wasn't totally sure of it's significance. Was there a version of this *with* parentheses that you'd intended to add?

Reply:
Yes - it's not used often, but the following is a valid test case:

```
@mock_s3()
def test_with_parentheses() -> int:
    assert boto3.client("s3").list_buckets()["Buckets"] == []
    return 123
```

Can you add that to this PR? It passes the MyPy check, so it's a good verification that all possible Moto usages are covered.

Reply:
makes sense, I'll have that change pushed up soon.